dev/com.ibm.ws.install/src/com/ibm/ws/install/maven/MavenDirectoryInstall.java

```diff
@@ -282,13 +269,28 @@ private String extractFeature(String filename) {
 
     }
 
-    private ArrayList<File> getSingleJsonPaths(File dir) throws IOException {
-        ArrayList<File> jsonFiles = new ArrayList<>();
-        try (Stream<Path> files = Files.walk(Paths.get(dir.toURI()))) {
-            jsonFiles.addAll(files.filter(f -> f.getFileName().toString().endsWith(".json")).map(f -> f.toFile())
-                    .collect(Collectors.toList()));
+    /**
+     * Populate found with the single json files
+     * 
+     * @param filepath
+     * @param found
+     */
+    private void getSingleJsonPaths(File filepath, ArrayList<File> found) {
+        File[] list = filepath.listFiles();
+        if (list == null) {
+            return;
+        }
+
+        for (File f : list) {
+            if (f.isDirectory()) {
+                getSingleJsonPaths(f, found);
```

Comment:
Recursive call is very inefficient.  Can we re-write it?

Reply:
Since we cannot use java 8 apis anymore, this will be a bit harder to deal with. I will look into ways to improve it

Reply:
Rewritten here https://github.com/OpenLiberty/open-liberty/pull/9032/commits/35a83338f9cb39902e9e4ca2c66a7dd4c78e9551